plugins/woocommerce/client/blocks/assets/js/atomic/utils/register-product-block-type.ts

```diff
@@ -171,6 +136,35 @@ export class BlockRegistrationManager {
 					}
 				} );
 
+				this.initialized = true;
+				// Site Editor Context (Templates, Patterns, etc.)
+			} else {
+				// Unsubscribe from the main subscription since we've detected our context
+				unsubscribe();
+
+				// getEditedPostSlug may return string or number so we cast it to string.
+				// @ts-expect-error getEditedPostSlug is not typed
+				const postSlug = String( editorSelectors.getEditedPostSlug() );
```

Comment:
I'm not sure that we should do this kind of cast, because:

```ts
String(undefined) === 'undefined' // true
```

Reply:
Hmm... I see what you mean. I was thinking if there's any "risk" of `String(undefined)` or `String(null)` and didn't think that actually post slug may potentially be or contain that string. In this specific function it still doesn't matter but I see your point!